### PR TITLE
Add some clarifications around C++/CLI projects

### LINKED ISSUE #33536: Doc needs to mention that C++/CLI .NET exe is not supported.
To migrate existing C++/CLI .NET Framework exe people need to create a c# .NET exe project which reference C++/CLI .NET dll.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 590dc11f-354e-ded2-469f-605febf6b394
* Version Independent ID: 097bfdc7-9d3d-852f-eba3-e7ea2e0c4239
* Content: [Migrating C++/CLI projects to .NET Core and .NET 5+ - .NET Core](https://learn.microsoft.com/en-us/dotnet/core/porting/cpp-cli)
* Content Source: [docs/core/porting/cpp-cli.md](https://github.com/dotnet/docs/blob/main/docs/core/porting/cpp-cli.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @mjrousos
* Microsoft Alias: **dotnetcontent**

### PR DESCRIPTION
**Pull request by adegeo**
## Summary

- Ran acrolinx.
- Removed references to .NET Core.
- Removed references to .NET Framework where it wasn't required.
- Cleaned up some phrases to make them simpler.
- Added info about C++/CLI executables not supported.

Fixes #33536 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/porting/cpp-cli.md](https://github.com/dotnet/docs/blob/86b0b24e45bd6477fe41e2b1d30cbaee6200bdf4/docs/core/porting/cpp-cli.md) | [How to port a C++/CLI project to .NET](https://review.learn.microsoft.com/en-us/dotnet/core/porting/cpp-cli?branch=pr-en-us-39228) |

### CODE REVIEW

File: docs/core/porting/cpp-cli.md [hunk 1]
```diff
@@ -1,52 +1,57 @@
...
-# How to port a C++/CLI project to .NET Core or .NET 5+
+# How to port a C++/CLI project to .NET
 
-Beginning with .NET Core 3.1 and Visual Studio 2019, [C++/CLI projects](/cpp/dotnet/dotnet-programming-with-cpp-cli-visual-cpp) can target .NET Core. This support makes it possible to port Windows desktop applications with C++/CLI interop layers to .NET Core/.NET 5+. This article describes how to port C++/CLI projects from .NET Framework to .NET Core 3.1.
+Beginning with Visual Studio 2019, [C++/CLI projects](/cpp/dotnet/dotnet-programming-with-cpp-cli-visual-cpp) can target .NET. This support makes it possible to port Windows desktop applications with C++/CLI interop layers to .NET. This article describes how to port C++/CLI projects from .NET Framework to .NET.
 
 ## C++/CLI .NET Core limitations
 
-There are some important limitations to porting C++/CLI projects to .NET Core compared to other languages:
+There are some important with C++/CLI projects and .NET compared to .NET Framework:
```
Comment (gewarren) [inline docs/core/porting/cpp-cli.md:14]:
```suggestion
There are some important limitations with C++/CLI projects and .NET compared to .NET Framework:
```

File: docs/core/porting/cpp-cli.md [hunk 2]
```diff
@@ -1,52 +1,57 @@
...
 author: mjrousos
-ms.date: 10/27/2023
+ms.date: 01/23/2024
 ---
 
-# How to port a C++/CLI project to .NET Core or .NET 5+
+# How to port a C++/CLI project to .NET
 
-Beginning with .NET Core 3.1 and Visual Studio 2019, [C++/CLI projects](/cpp/dotnet/dotnet-programming-with-cpp-cli-visual-cpp) can target .NET Core. This support makes it possible to port Windows desktop applications with C++/CLI interop layers to .NET Core/.NET 5+. This article describes how to port C++/CLI projects from .NET Framework to .NET Core 3.1.
+Beginning with Visual Studio 2019, [C++/CLI projects](/cpp/dotnet/dotnet-programming-with-cpp-cli-visual-cpp) can target .NET. This support makes it possible to port Windows desktop applications with C++/CLI interop layers to .NET. This article describes how to port C++/CLI projects from .NET Framework to .NET.
```
Comment (gewarren) [inline docs/core/porting/cpp-cli.md:10]:
```suggestion
Beginning with Visual Studio 2019, [C++/CLI projects](/cpp/dotnet/dotnet-programming-with-cpp-cli-visual-cpp) can target .NET. This support makes it possible to port Windows desktop applications with C++/CLI interop layers from .NET Framework to .NET. This article describes how to port C++/CLI projects from .NET Framework to .NET.
```

File: docs/core/porting/cpp-cli.md [hunk 3]
```diff
@@ -1,52 +1,57 @@
...
+There are some important with C++/CLI projects and .NET compared to .NET Framework:
 
-- C++/CLI support for .NET Core is Windows only.
-- C++/CLI projects can't target .NET Standard, only .NET Core or .NET Framework.
-- C++/CLI projects don't support the new SDK-style project file format. Instead, even when targeting .NET Core, C++/CLI projects use the existing *.vcxproj* file format.
-- C++/CLI projects can't multitarget multiple .NET platforms. If you need to build a C++/CLI project for both .NET Framework and .NET Core, use separate project files.
-- .NET Core doesn't support `-clr:pure` or `-clr:safe` compilation, only the new `-clr:netcore` option (which is equivalent to `-clr` for .NET Framework).
+- C++/CLI support for .NET is Windows only.
+- C++/CLI projects can't target .NET Standard.
+- C++/CLI projects don't support the new SDK-style project file format. Instead, C++/CLI projects use the existing _.vcxproj_ file format.
```
Comment (gewarren) [inline docs/core/porting/cpp-cli.md:18]:
```suggestion
- C++/CLI projects don't support the newer SDK-style project file format. Instead, C++/CLI projects use the _.vcxproj_ file format.
```

File: docs/core/porting/cpp-cli.md [hunk 4]
```diff
@@ -1,52 +1,57 @@
...
+- C++/CLI projects can't target multiple .NET platforms. If you need to build a C++/CLI project for both .NET and .NET Framework.
```
Comment (gewarren) [inline docs/core/porting/cpp-cli.md:19]:
```suggestion
- C++/CLI projects can't target multiple .NET platforms. If you need to build a C++/CLI project for both .NET and .NET Framework, use separate project files.
```

File: docs/core/porting/cpp-cli.md [hunk 5]
```diff
@@ -1,52 +1,57 @@
...
+- .NET doesn't support `-clr:pure` or `-clr:safe` compilation, only the new `-clr:netcore` option (which is equivalent to `-clr` for .NET Framework).
```
Comment (gewarren) [inline docs/core/porting/cpp-cli.md:20]:
```suggestion
- .NET doesn't support `-clr:pure` or `-clr:safe` compilation, only the newer `-clr:netcore` option (which is equivalent to `-clr` for .NET Framework).
```

**Review by gewarren — APPROVED** [review]

File: docs/core/porting/cpp-cli.md (hunk 3 above)
Comment (olgaark) [reply docs/core/porting/cpp-cli.md:18]: It is better to say that C++/CLI projects use the same vcxproj file format as other vc projects, no changes are needed there. The appropriate .NET SDK files are automatically imported when C++/CLI project targets .NET.

File: docs/core/porting/cpp-cli.md (hunk 4 above)
Comment (olgaark) [reply docs/core/porting/cpp-cli.md:19]: Same comment for "C++/CLI projects don't support the new SDK-style project file format. Instead, C++/CLI projects use the existing _.vcxproj_ file format."

File: docs/core/porting/cpp-cli.md (hunk 3 above)
Comment (olgaark) [reply docs/core/porting/cpp-cli.md:18]: You also need to add that .exe (application) is not supported, only dll (library).

**Pull request merged**